docs.ps1

```diff
@@ -0,0 +1,53 @@
+param([Switch]$D = $false)
+
+# Check working directory is the root directory of the PDKGO repository
+if ((Test-Path -Path ".\docs\md") -eq $false)
+{
+    Write-Host 'Please run this script from the root directory of the PDKGO project.'
+    Exit 1
+}
+# Check hugo extended is installed
+if ((hugo version |Select-String -Pattern "extended") -eq $null)
+{
+    Write-Host 'The "extended" version of hugo was not found, please install it.'
+    Exit 1
+}
+# Check git is installed
+if ((Get-Command "git.exe" -ErrorAction SilentlyContinue) -eq $null)
+{
+   Write-Host "Git version control was not found, please install it."
+   Exit 1
+}
+# Check if npm is installed
+if ((Get-Command "npm.cmd" -ErrorAction SilentlyContinue) -eq $null)
+{
+   Write-Host "NPM was not found, please install it."
+   Exit 1
+}
+
+# Check if the docs site directory exists
+if ((Test-Path -Path ".\docs\site") -eq $false)
+{
+    git clone https://github.com/puppetlabs/devx.git docs\site
+    cd docs\site
```

Comment:
Instead of `cd`, which is an alias for `Set-Location`, prefer using either `Set-Location` or `Push-Location` - the advantage of the latter being that you can later call `Pop-Location` to return to your initial path if needed.